core/src/main/java/org/sonarsource/sonarlint/core/container/global/MetadataLoader.java

```diff
@@ -28,20 +28,32 @@
 
 public class MetadataLoader {
 
-  private static final String VERSION_FILE_PATH = "/sonar-api-version.txt";
+  private static final String SONAR_PLUGIN_API_VERSION_FILE_PATH = "/sonar-api-version.txt";
+  private static final String SONARLINT_PLUGIN_API_VERSION_FILE_PATH = "/sonarlint-api-version.txt";
 
   private MetadataLoader() {
     // only static methods
   }
 
-  public static Version loadVersion(System2 system) {
-    URL url = system.getResource(VERSION_FILE_PATH);
+  public static Version loadSonarPluginApiVersion() {
+    URL url = System2.INSTANCE.getResource(SONAR_PLUGIN_API_VERSION_FILE_PATH);
 
     try (Scanner scanner = new Scanner(url.openStream(), StandardCharsets.UTF_8.name())) {
       String versionInFile = scanner.nextLine();
       return Version.parse(versionInFile);
     } catch (IOException e) {
-      throw new IllegalStateException("Can not load " + VERSION_FILE_PATH + " from classpath ", e);
+      throw new IllegalStateException("Can not load " + SONAR_PLUGIN_API_VERSION_FILE_PATH + " from classpath ", e);
+    }
+  }
+
+  public static Version loadSonarLintPluginApiVersion() {
+    URL url = MetadataLoader.class.getResource(SONARLINT_PLUGIN_API_VERSION_FILE_PATH);
+
+    try (Scanner scanner = new Scanner(url.openStream(), StandardCharsets.UTF_8.name())) {
```

Comment:
Nitpick: this could probably be extracted to a utility method, only the resource path changes.